arch/arm/src/nrf52/nrf52_gpiote.c

```diff
@@ -135,6 +141,61 @@ static int nrf52_gpiote_isr(int irq, FAR void *context, FAR void *arg)
         }
     }
 
+  /* Check for PORT event */
+
+  regval = nrf52_gpiote_getreg(NRF52_GPIOTE_EVENTS_PORT_OFFSET);
+  if (regval)
+    {
+      uint32_t addr = 0;
+
+      /* Ack PORT event */
+
+      nrf52_gpiote_putreg(NRF52_GPIOTE_EVENTS_PORT_OFFSET, 0);
+
+      /* For each GPIO port, get LATCH register */
+
+      for (i = 0; i < NRF52_GPIO_NPORTS; i++)
+        {
+          switch (i)
+            {
+              case 0:
+                addr = NRF52_GPIO_P0_BASE + NRF52_GPIO_LATCH_OFFSET;
+              break;
+#ifdef CONFIG_NRF52_HAVE_PORT1
+              case 1:
+                addr = NRF52_GPIO_P1_BASE + NRF52_GPIO_LATCH_OFFSET;
+              break;
+#endif
+            }
+
+          /* Retrieve LATCH register */
+
+          regval = getreg32(addr);
+
+          /* Clear LATCH register (this may set PORT again) */
+
+          putreg32(0xffffffff, addr);
+
+          /* Check for pins with DETECT bit high in LATCH register
+           * and dispatch callback if set
+           */
+
+          for (j = 0; j < NRF52_GPIO_NPINS; j++)
```

Comment:
Scanning all pins is very ineffective, especially when we have more than 1 port.
It would be better to store the pin and port for a given event and compare only these.

Reply:
Yes, I can improve this. I was looking for some non-looping code to only go through set bits in LATCH register. This way it does only what is needed and does nothing when LATCH is zero.

Reply:
I didn't find a non-looping approach to only go over bits set. I will ultimately requires looping to find the position of next bit set so I don't think it helps much. I will just add a check to see if the register is zero and not look for bits set then.